src/cli.rs

```diff
@@ -8,4 +11,26 @@ pub struct Cli {
 
    #[clap(flatten)]
    pub render: crate::backend::cli::RendererCli,
+
+   #[command(subcommand)]
+   pub command: Option<Commands>,
+}
+
+// Borrow checker complains about lifetimes when we use RoomId::try_from
+// for validation, despite it being compatible type.
+// So just wrap it in a function.
+fn is_valid_room_id(value: &str) -> Result<RoomId, RoomIdError> {
+   RoomId::try_from(value)
+}
```

Comment:
IIRC it should be possible to implement [`std::str::FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) and clap will use that without you having to specify `value_parser` and wrap it in a string.

For reference: https://docs.rs/clap/latest/clap/macro.value_parser.html

IMHO it's a mistake that `RoomId` implements `TryFrom<&str>` in the first place, it should implement `FromStr` instead.